java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java

```diff
@@ -1515,6 +1520,12 @@ public static PackageAndClassName parseFromQualifiedName(String qualifiedName)
             }
             return new PackageAndClassName(packageName, className);
         }
+
+        private static PackageAndClassName parseFromQualifiedNameWithConstant(String qualifiedName)
+        {
+            int lastDot = qualifiedName.lastIndexOf('.');
```

Comment:
Not null validation ?

Reply:
Good point :+1: 
Food for thought on which should be the way to manage the case when `qualifiedName` is null if i add the null validation.
Any suggestion?

- force a reasonable default (e.g. empty string) for `qualifiedName` would lead to an "empty" `PackageAndClassName` result which seems useless
- on the other side, having the NPE could make sense: the caller must provide a not-null (and not-empty?) `qualifiedName` value

WDYT?

Reply:
If `qualifiedName` is *null* the problem is not only here but also in `parseFromQualifiedName` method .

Reply:
I think to keep this PR aligned with its original scope and do not break any current features/behaviour in windup, 
it seems clear to me that, no matter this change, the previous code was failing with NPE if `qualifiedName` was null [1]
So I can see your point about the NPE but it seems the fix (if any) is going out of scope for this PR.

[1] https://github.com/windup/windup/blob/master/java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java#L1497

Reply:
Checking the code I see practically no way this `qualifiedName` could be null. So although it would be good to have this check, I agree with you about the scope of this issue.